app/api/client/create/route.ts

```diff
@@ -0,0 +1,78 @@
+import dbConnect from "@/app/lib/dbConnect";
+import mongoose from "mongoose";
+import Client from "@/app/models/Client";
+import { ObjectId } from "mongoose";
+import bcrypt from "bcrypt";
+
+interface ClientModelInterface {
```

Comment:
Change Interface declaration to the following:

interface ClientModelInterface → interface ClientModelInterface extends Document

The reason for this is because `Client.create()` returns a Mongoose `Document` that contains properties given by the model, in this case `Client`. This lets TypeScript know that this interface must include the functionalities of a Mongoose `Document`.